### PR TITLE
Integrate support for Sentry reporting

### LINKED ISSUE #120: Integrate Sentry support

### PR DESCRIPTION
**Pull request by 4lovi4**
## What do these changes do?
sentry integration template for the generated app

## Related issue number

Closes #120

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py [hunk 1]
```diff
@@ -2,8 +2,10 @@
 import uvloop
 {%- endif %}
 from aiohttp import web
-
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py]: Why did you remove this section separator? Get it back!

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py [hunk 2]
```diff
@@ -2,8 +2,10 @@
...
 from .app import init_app
+    {%- if cookiecutter.use_sentry == 'y' %}
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py]: Don't add trailing spaces.
```suggestion
{%- if cookiecutter.use_sentry == 'y' %}
```

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py [hunk 3]
```diff
@@ -3,16 +3,19 @@
 
 {%- if cookiecutter.use_redis == 'y' %}
 from functools import partial
+import aioredis
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py]: This change is unrelated. And also this import doesn't belong to the section with stdlib imports.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py [hunk 4]
```diff
@@ -3,16 +3,19 @@
...
 import aiopg.sa
 {%- endif %}
 from aiohttp import web
-{%- if cookiecutter.use_redis == 'y' %}
-import aioredis
+{%- if cookiecutter.use_sentry == 'y' %}
+import sentry_sdk
+from sentry_sdk.integrations.aiohttp import AioHttpIntegration
 {%- endif %}
+
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py]: No need to separate different third-party libs.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py [hunk 5]
```diff
@@ -78,6 +81,19 @@ def init_jinja2(app: web.Application) -> None:
...
+{%- if cookiecutter.use_sentry == 'y' %}
+
+
+def init_sentry(app: web.Application) -> None:
+    '''
+    Initialize sentry sdk for aiohttp
+    '''
+    config = app['config']['sentry']
+    sentry_sdk.init(
+        dsn=f'https://{configp["key"]}@sentry.io/{config["project"]}',
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py]: Nope. User should provide the complete DSN via the config. Not a separate key. They may have an on-premise install of it and so on. Don't steal the control of this from them.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py [hunk 6]
```diff
@@ -36,6 +36,13 @@
             'port': trafaret.Int(),
         }),
     {%- endif %}
+{%- if cookiecutter.use_sentry == 'y' %}
+    trafaret.Key('sentry'):
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py]: No. Accept a string for dsn.

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py (hunk 1 above)
Comment (4lovi4) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py]: I've got it back

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py (hunk 2 above)
Comment (4lovi4) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py]: trailing is deleted

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py (hunk 3 above)
Comment (4lovi4) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py]: Got it.  Reverted it back.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py (hunk 4 above)
Comment (4lovi4) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py]: ok

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py (hunk 5 above)
Comment (4lovi4) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py]: Chaged for the dsn key in config template.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py (hunk 6 above)
Comment (4lovi4) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py]: I've put the dsn key here.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py [hunk 7]
```diff
@@ -78,6 +82,19 @@ def init_jinja2(app: web.Application) -> None:
...
+
+
+def init_sentry(app: web.Application) -> None:
+    '''
+    Initialize sentry sdk for aiohttp
+    '''
+    config = app['config']['sentry']
+    sentry_sdk.init(
+        dsn=config["dsn"],
+        integrations=[AioHttpIntegration()]
```
Comment (webknjaz) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/app.py]: Maybe also set `environment`, `server_name`, `attach_stacktrace` and `release`?

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py (hunk 6 above)
Comment (webknjaz) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py]: Either add other options or turn this into a scalar option. No need for mapping if it's just one value.

**Review by webknjaz — APPROVED** [review]
LGTM overall. Let's see what others think.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py [hunk 8]
```diff
@@ -22,6 +25,11 @@ def main() -> None:
     {%- if cookiecutter.use_uvloop == 'y' %}
     uvloop.install()
     {%- endif %}
+
+    {%- if cookiecutter.use_sentry == 'y' %}
+    init_sentry(app)
+    {%- endif %}
+
```
Comment (Arfey) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py]: This is simple init function, so move initialize `sentry` to `init_app` from [app.py](https://github.com/aio-libs/create-aio-app/blob/master/create_aio_app/template/%7B%7Bcookiecutter.project_name%7D%7D/%7B%7Bcookiecutter.project_name%7D%7D/app.py#L86-L88)

Insert this function after `init_routes(app)`, please. I guess so this features is not useful for developer mode. As a solution, I recommend provide some value to config file. Some like `debug` with `True/False` or `mode` with `prd/test/dev` properties. (In my opinion `mode` will be more useful because give possibility split different environments) And after that use `sentry` only for production.

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py [hunk 9]
```diff
@@ -36,6 +36,12 @@
...
+        trafaret.Dict({
+            'dsn': trafaret.String(),
+        }),
+    {%- endif %}
```
Comment (Arfey) [inline create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/utils/common.py]: After generation a new project with `sentry`, I will get error that my config file doesn't have `sentry` key. Am i right? If it's `true` so add this keys to `config` files, please.

**Review by Arfey — CHANGES_REQUESTED** [review]

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py (hunk 8 above)
Comment (webknjaz) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py]: I usually set dev env and also a specifier of my laptop, so that if errors come from different devs it could be easy to recognize.

**Comment by Arfey** [comment]
It's look as very useful feature for a lot of people. If we want that people use  this then we need to describe about that for them. 

- [ ] add description about this flag to the `readme`
- [ ] add description about setting sentry to the documentation

@4lovi4 can you do that? 😀

File: create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py (hunk 8 above)
Comment (Arfey) [reply create_aio_app/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/__main__.py]: I don't use sentry so can't  say anything about best usage this library and just proposed.

**Comment by 4lovi4** [comment]
Going to refresh the doc.

**Comment by webknjaz** [comment]
@4lovi4 any updates?

**Comment by Arfey** [comment]
@4lovi4 any updates `2.0`?)
